model/Index/Service/RecreatingIndexService.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2025 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\model\Index\Service;
+
+use Elastic\Elasticsearch\Client;
+use Exception;
+use oat\oatbox\reporting\Report;
+use oat\tao\model\AdvancedSearch\AdvancedSearchChecker;
+use oat\taoAdvancedSearch\model\SearchEngine\Service\IndexPrefixer;
+
+class RecreatingIndexService
```

Comment:
- [ ] Lets delete this class please https://github.com/oat-sa/extension-tao-advanced-search/pull/140/files#r2012394864